src/Microsoft.Kusto.ServiceLayer/DataSource/ReliableDataSourceConnection.cs

```diff
@@ -211,10 +211,7 @@ public Task OpenAsync(CancellationToken token)
             // Check if retry policy was specified, if not, disable retries by executing the Open method using RetryPolicy.NoRetry.
             try
             {
-                return _connectionRetryPolicy.ExecuteAction(async () =>
```

Comment:
This change will be moving into a different PR as per Karl's request. I'll revisit why this change was made since off the top of my head I can't remember why.